app/test/package/upload_test.dart

```diff
@@ -1049,6 +1049,32 @@ void main() {
         );
       });
 
+      testWithProfile('has dependency does not exist', fn: () async {
+        final tarball = await packageArchiveBytes(
+            pubspecContent:
+                generatePubspecYaml('xyz', '1.0.0') + '  abc: ^1.0.0\n');
+        final rs = createPubApiClient(authToken: adminClientToken)
+            .uploadPackageBytes(tarball);
+        await expectApiException(
+          rs,
+          status: 400,
+          code: 'PackageRejected',
+          message: 'Dependency `abc` does not exist.',
+        );
+      });
+
+      testWithProfile('has SDK-dependency', fn: () async {
+        final tarball = await packageArchiveBytes(
+            pubspecContent: generatePubspecYaml('xyz', '1.2.3') +
+                '  my_sdk_dep:\n'
+                    '    sdk: dart\n');
```

Comment:
This will allow package to have any dependency as long as they have `sdk: dart` (or `sdk:flutter`)?